File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/datasync/channel/RemoveChannel.java
package com.pubnub.api.java.endpoints.datasync.channel;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.java.models.consumer.datasync.channel.PNDataSyncRemoveChannelResult;
import org.jetbrains.annotations.Nullable;

/**
 * @see com.pubnub.api.java.datasync.DataSync#removeChannel(String)
 */
public interface RemoveChannel extends Endpoint<PNDataSyncRemoveChannelResult> {
    /**
     * Optional eTag for a conditional delete ({@code If-Match} header).
     */
    RemoveChannel ifMatch(@Nullable String ifMatch);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/datasync/channel/RemoveChannel.kt
package com.pubnub.api.endpoints.datasync.channel

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.datasync.channel.DataSyncRemoveChannelResult

/**
 * @see [com.pubnub.api.datasync.DataSync.removeChannel]
 */
actual interface RemoveChannel : Endpoint<DataSyncRemoveChannelResult>
